zio-http/src/main/scala/zhttp/service/ServerStart.scala

```diff
@@ -0,0 +1,31 @@
+package zhttp.service
+
+sealed trait ServerStart { self =>
```

Comment:
Add scala doc for this class.

Reply:
If we are sticking with a product type for now, we should definitely define it within Server.scala. I'll move Start there and add the doc.